pyp_kcwi/sub_sky.py

```diff
@@ -0,0 +1,142 @@
+# This function is used to subtract sky median
+#
+# INPUT:     - sky_file
+#            - img_file
+#            - mask_limit    mask limit
+#            - cut_ch        cut channel number
+#
+# INPUT EXAMPLE:
+#   median('kb171021_00083_icuber.fits', 'kb171021_00082_icuber.fits', 40, cut_ch=300)
+#
+# MODIFICATION HISTORY:
+#   Written by: Qiong Li
+#   2017-11-01 Initial version
+
+def median(sky_file, img_file, mask_limit, cut_ch=300):
+    from astropy.io import fits
+    import numpy as np
+    import matplotlib.pyplot as plt
+    import matplotlib as mpl
+    sky_input, sh = fits.getdata(sky_file, header=True)
+    img_input, ih = fits.getdata(img_file, header=True)
+
+    img=img_input[cut_ch:(len(img_input[:,1,1])-cut_ch),:,:]
+    sky=sky_input[cut_ch:(len(img_input[:,1,1])-cut_ch),:,:]
+
+    ### begin to mask qso on sky
+    # define final mask image
+    mask_sum = np.zeros(sky[1,:,:].shape)
+
+    ## sky[w,x,y]; for every wavelength: len(sky[:,1,1])
+    for j in np.arange(0, len(sky[:,1,1]), 1):
+
+        # sky image for each channel
+        s=sky[j,:,:]
+
+        # define mask image for each channel
+        mask = np.zeros(sky[1,:,:].shape)
+
+    # begin 10 loops
```

Comment:
Can we use astropy sigma_clip:

http://docs.astropy.org/en/stable/api/astropy.stats.sigma_clip.html

Reply:
Done. But become slow...